### PR TITLE
Top uploaders

### PR DESCRIPTION
**Pull request by prakhar-s**
aggregated on author ids to get counts for top uploaders and then used esmodule to get fetch other details using these ids

### CODE REVIEW

File: src/main/java/com/strandls/observation/pojo/TopUploadersInfo.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.strandls.observation.pojo;
+
+public class TopUploadersInfo {
+	
+	String name;
```
Comment (akrudra) [inline src/main/java/com/strandls/observation/pojo/TopUploadersInfo.java:5]: add private access modifiers to all the variables used

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java [hunk 2]
```diff
@@ -511,6 +526,161 @@ public MapAggregationResponse mapAggregate(String index, String type, String sGr
...
+				Long authorId = allUploadersInfo.get(k).getAuthorId();
+
+				TopUploadersInfo tempUploader = new TopUploadersInfo(name, pic, authorId, counts.get(k));
+				uploadersResult.add(tempUploader);
+
+			}
+
+			aggregationStatsResponse.setGroupTopUploaders(uploadersResult);
+		} catch (ApiException e) {
+			// TODO Auto-generated catch block e.printStackTrace(); }
```
Comment (akrudra) [inline src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java:677]: add a logger
and a exception catch

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java [hunk 3]
```diff
@@ -511,6 +526,161 @@ public MapAggregationResponse mapAggregate(String index, String type, String sGr
...
+		}
+		int uploadersSize = uploadersoffset + 10;
+		int uploadersCount = 1;
+		String authorIds = "";
+
+		List<Long> counts = new ArrayList<>();
+
+		Map<String, Long> uploaders = mapAggStatsResponse.get("author_id").getGroupAggregation();
+
+		if (user != null && !user.isEmpty()) {
```
Comment (akrudra) [inline src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java:631]: try to write extraction logic in  private function and  call it when req, instead writing inside this block

**Review by akrudra — CHANGES_REQUESTED** [review]

File: src/main/java/com/strandls/observation/pojo/TopUploadersInfo.java (hunk 1 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/observation/pojo/TopUploadersInfo.java]: added private

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java (hunk 2 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java]: added logger

File: src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java (hunk 3 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/observation/service/Impl/ObservationListServiceImpl.java:639]: added a private function for extraction logic

**Review by akrudra — APPROVED** [review]
remove unsed imports and warning and  organize and format the code

File: src/main/java/com/strandls/observation/pojo/TopIdentifiersInfo.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+public class TopIdentifiersInfo {
+	
+	private String name;
+	private String pic;
+	private Long authorId;
+	private Long count;
+	public TopIdentifiersInfo() {
+		super();
+		// TODO Auto-generated constructor stub
```
Comment (akrudra) [inline src/main/java/com/strandls/observation/pojo/TopIdentifiersInfo.java:11]: remove the TODO and apply proper spacing

**Review by akrudra — APPROVED** [review]
please double-check for warnings, unused code, imports, and formating

**Pull request merged**